### PR TITLE
distinct support in sea-orm

### LINKED ISSUE #275: Distinct support?
I saw sea-query supports it, but It doesn't seem like it's supported yet in sea-orm.

### PR DESCRIPTION
**Pull request by kyoto7250**
close #275 

## Adds

This PR adds `distinct` methods in `helper.rs`

- [x]  distinct
- [x] distinct on (only `sqlx-postgres`)

### CODE REVIEW

File: src/query/helper.rs [hunk 1]
```diff
@@ -149,6 +152,61 @@ pub trait QuerySelect: Sized {
...
+    ///         .distinct_on(cake::Column::Name)
+    ///         .build(DbBackend::Postgres)
+    ///         .to_string(),
+    ///     "SELECT DISTINCT ON (\"name\") \"cake\".\"id\", \"cake\".\"name\" FROM \"cake\" WHERE \"cake\".\"name\" LIKE '%cheese%'"
+    /// );
+    /// ```
+    #[cfg(feature = "sqlx-postgres")]
+    fn distinct_on<C>(mut self, col: C) -> Self
+    where
+        C: IntoColumnRef,
```
Comment (billy1624) [inline src/query/helper.rs:204]: Following the method signature of SeaQuery's method, we could take a iterator here. e.g.
```rs
pub fn distinct_on<T, I>(mut self, cols: I) -> Self
    where
        T: IntoColumnRef,
        I: IntoIterator<Item = T>
{ ... }
```

File: src/query/helper.rs [hunk 2]
```diff
@@ -149,6 +152,61 @@ pub trait QuerySelect: Sized {
...
+    ///         .filter(
+    ///             Condition::all().add_option(input.name.map(|n| cake::Column::Name.contains(&n)))
+    ///         )
+    ///         .distinct_on(cake::Column::Name)
+    ///         .build(DbBackend::Postgres)
+    ///         .to_string(),
+    ///     "SELECT DISTINCT ON (\"name\") \"cake\".\"id\", \"cake\".\"name\" FROM \"cake\" WHERE \"cake\".\"name\" LIKE '%cheese%'"
+    /// );
+    /// ```
+    #[cfg(feature = "sqlx-postgres")]
```
Comment (billy1624) [inline src/query/helper.rs:201]: We try to avoid feature gated method

File: src/query/helper.rs [hunk 3]
```diff
@@ -8,6 +8,9 @@ use sea_query::{
 };
 pub use sea_query::{Condition, ConditionalStatement, DynIden, JoinType, Order, OrderedStatement};
 
+#[cfg(feature = "sqlx-postgres")]
```
Comment (billy1624) [inline src/query/helper.rs:11]: This line would be unnecessary if we dropped the feature gate below

**Review by billy1624 — COMMENTED** [review]
Hey @kyoto7250, thanks for contributing!! Look pretty good. I've some comments
Please check and see if it make sense.
Thanks!! :)

File: src/query/helper.rs (hunk 1 above)
Comment (kyoto7250) [reply src/query/helper.rs]: I fixed in https://github.com/SeaQL/sea-orm/pull/902/commits/db1111c0e5c5935b7e0782234906b741490628ff

File: src/query/helper.rs (hunk 2 above)
Comment (kyoto7250) [reply src/query/helper.rs]: I fixed in https://github.com/SeaQL/sea-orm/pull/902/commits/c9871cb2d25de92336501799faa7eb29595ef95b

File: src/query/helper.rs (hunk 1 above)
Comment (billy1624) [reply src/query/helper.rs]: Thanks!! @kyoto7250

**Review by billy1624 — APPROVED** [review]
Thanks again for the great contributions! @kyoto7250

**Pull request merged**